<?xml version="1.0" ?>
<launch>
    <arg name="eye_on_hand" default="true" />

    <!-- these parameters depend on the robot; you can copy these for the Franka Panda -->
    <arg name="move_group" default="panda_arm" />
    <arg name="move_group_namespace" value="/easy_handeye_eye_on_hand" />
    <arg name="robot_base_frame" value="panda_link0"/>
    <arg name="robot_effector_frame" value="panda_link7"/>

    <!-- these parameters depend on the tracking system; these are only valid for this simulator, you'll have to find out yours -->
    <!-- <arg name="tracking_base_frame" value="rgb_camera_link"/> -->
    <arg name="tracking_base_frame" value="camera_color_optical_frame"/>
    <arg name="tracking_marker_frame" value="aruco_tag"/>

    <arg name="markerId"        default="582"/>
    <arg name="markerSize"      default="0.034"/>    <!-- in m -->
    <arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->

    <include file="$(find realsense2_camera)/launch/rs_camera.launch" />
    <!-- <include file="$(find azure_kinect_ros_driver)/launch/driver.launch" /> -->

    <!-- <group ns="rgb" > -->
    <group ns="/camera/color" >
      <node pkg="image_proc" type="image_proc" name="image_proc"/>
    </group>

    <node pkg="aruco_ros" type="single" name="aruco_single">
        <remap from="/camera_info" to="/camera/color/camera_info" />
        <!-- <remap from="/camera_info" to="/rgb/camera_info" /> -->
        <!-- <remap from="/image" to="/camera/color/image_raw" /> -->
        <remap from="/image" to="/camera/color/image_rect_color" />
        <!-- <remap from="/image" to="/rgb/image_rect_color" /> -->
        <param name="image_is_rectified" value="true"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value=""/>  
        <param name="camera_frame"       value="camera_color_optical_frame"/>
        <param name="marker_frame"       value="aruco_tag" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node>

    <!-- start easy_handeye calibration procedure -->
    <include file="$(find easy_handeye)/launch/calibrate.launch" >
        <arg name="start_rviz" value="false" />
        <arg name="publish_dummy" value="false" />

        <arg name="move_group" value="$(arg move_group)" />
        <arg name="move_group_namespace" value="$(arg move_group_namespace)" />

        <arg name="eye_on_hand" value="$(arg eye_on_hand)" />
        <arg name="tracking_base_frame" value="$(arg tracking_base_frame)" />
        <arg name="tracking_marker_frame" value="$(arg tracking_marker_frame)" />
        <arg name="robot_base_frame" value="$(arg robot_base_frame)" />
        <arg name="robot_effector_frame" value="$(arg robot_effector_frame)" />

        <arg name="freehand_robot_movement" value="true" />
        <arg name="robot_velocity_scaling" value="0.5" />
        <arg name="robot_acceleration_scaling" value="0.2" />
    </include>

    <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
       <remap from="image" to="/aruco_single/result" />
    </node>

</launch>
